### PR TITLE
move serializer

### PR DESCRIPTION
**Pull request by xuke444**
Signed-off-by: Ke Xu <xuke@microsoft.com>

### CODE REVIEW

File: responsibleai/responsibleai/_interfaces.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    EBM = "EBM"
+    blackbox = "blackbox"
+
+
+class ModelMethod(str, Enum):
+    classifier = "classifier"
+    regressor = "regressor"
+
+
+class ModelExplanationData(object):
```
Comment (gregorybchris) [inline responsibleai/responsibleai/_interfaces.py:54]: Nit: in Python 3 you don't need `(object)`

File: responsibleai/responsibleai/_interfaces.py [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    ci_upper: float
+    feature: str
+    p_value: float
+    point: float
+    stderr: float
+    zstat: float
+
+
+class CausalData(object):
+    globalCausalEffects: List[CausalMetric]
```
Comment (gregorybchris) [inline responsibleai/responsibleai/_interfaces.py:79]: All python objects should use snake case `global_causal_effects`

Comment (gregorybchris) [reply responsibleai/responsibleai/_interfaces.py:79]: I didn't comment this when this code was in raiwidgets, but now that it's in responsibleai it should be proper python style

File: responsibleai/responsibleai/_interfaces.py [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+
+class ModelClass(str, Enum):
+    Tree = "Tree"
+    EBM = "EBM"
+    blackbox = "blackbox"
+
+
+class ModelMethod(str, Enum):
+    classifier = "classifier"
```
Comment (gregorybchris) [inline responsibleai/responsibleai/_interfaces.py:50]: Enum elements should be upper snake case `CLASSIFIER`/`REGRESSOR`

File: responsibleai/responsibleai/_interfaces.py [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+    localFeatureImportance: FeatureImportance
+    globalFeatureImportance: FeatureImportance
+    ebmGlobalExplanation: EBMGlobalExplanation
+    customVis: str
+
+
+class ModelClass(str, Enum):
+    Tree = "Tree"
+    EBM = "EBM"
+    blackbox = "blackbox"
```
Comment (gregorybchris) [inline responsibleai/responsibleai/_interfaces.py:46]: `blackbox` -> `BLACKBOX`

File: responsibleai/responsibleai/_interfaces.py [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+class BoundedCoordinates(object):
+    type: str
+    names: List
+    scores: List
+    scores_range: List
+    upper_bounds: List
+    lower_bounds: List
+
+
+class EBMGlobalExplanation (object):
```
Comment (gregorybchris) [inline responsibleai/responsibleai/_interfaces.py:26]: space between class name and `(`

File: responsibleai/responsibleai/_input_processing.py [hunk 6]
```diff
@@ -20,11 +20,14 @@ def _convert_to_list(array):
             raise ValueError("Exceeds maximum number of features for "
                              "visualization (1000)")
         return array.toarray().tolist()
-
-    if (isinstance(array, pd.DataFrame) or isinstance(array, pd.Series)):
+    if (isinstance(array, pd.DataFrame)):
```
Comment (gregorybchris) [inline responsibleai/responsibleai/_input_processing.py:23]: remove conditional parens here and below also

File: responsibleai/responsibleai/_internal/constants.py [hunk 7]
```diff
@@ -44,3 +44,21 @@ class ErrorAnalysisManagerKeys(object):
...
+
+    EXAMPLES = 'examples'
+    LABELS = 'labels'
+    PREDICTIONS = 'predictions'
+    PREDICT_PROBA = 'predict_proba'
+
+
+class ExplanationKeys(object):
+
+    MLI_EBM_GLOBAL_EXPLANATION_KEY = "ebm_global"
```
Comment (gregorybchris) [inline responsibleai/responsibleai/_internal/constants.py:60]: Nit: enum string literals are usually single quoted as they are more symbols than raw text

File: responsibleai/responsibleai/_managers/explainer_manager.py [hunk 8]
```diff
@@ -155,6 +160,109 @@ def list(self):
...
+    def get_data(self):
+        """Get explanation data
+
+        :return: A array of ModelExplanationData.
+        :rtype: List[ModelExplanationData]
+        """
+        return [
+            self._get_interpret(i) for i in self.get()]
+
+    def _get_interpret(self, explanation):
```
Comment (gregorybchris) [inline responsibleai/responsibleai/_managers/explainer_manager.py:172]: In order to make this more unit testable, what do you think about introducing an `explainer_serializer.py` with an `ExplainerSerializer` object that will encapsulate the conversion from explanation to JSON-serializable data? @imatiach-msft @xuke444

File: responsibleai/responsibleai/_internal/constants.py [hunk 9]
```diff
@@ -44,3 +44,21 @@ class ErrorAnalysisManagerKeys(object):
...
+    MLI_EXPLANATION_DATASET_KEY = "evaluation_dataset"
+    MLI_EXPLANATION_TYPE_KEY = "explanation_type"
+    MLI_GLOBAL_EXPLANATION_KEY = "global_feature_importance"
+    MLI_LOCAL_EXPLANATION_KEY = "local_feature_importance"
```
Comment (gregorybchris) [inline responsibleai/responsibleai/_internal/constants.py:64]: "MLI" and "KEY" are a bit redundant here if we can always use ExplanationKeys.LOCAL_EXPLANATION

Comment (gregorybchris) [reply responsibleai/responsibleai/_internal/constants.py:64]: Also @imatiach-msft maybe I should know this, but what does MLI mean?

File: responsibleai/responsibleai/_managers/explainer_manager.py (hunk 8 above)
Comment (xuke444) [reply responsibleai/responsibleai/_managers/explainer_manager.py:172]: will use a separate pr for this.

**Comment by gregorybchris** [comment]
Approving with the assumption that these comments will be addressed in a followup PR 🙂

File: responsibleai/responsibleai/_interfaces.py (hunk 2 above)
Comment (xuke444) [reply responsibleai/responsibleai/_interfaces.py:79]: There will be hundreds of file changed on ui side, will send a separate pr for it today.

**Review by gregorybchris — APPROVED** [review]

**Pull request merged**